### ISSUE TITLE
UnboundLocalError: local variable 'pad' referenced before assignment

### ISSUE THREAD

**Issue by jklenzing** [MAVENSDC/cdflib#210]
Working through the supported instruments at pysatNASA with the new cdflib.  Found an issue with loading some of the older datasets like DE2.  These files are flagged with version 2.7.2 in `cdf_info`.  Newer datasets (such as C/NOFS, using version 3.2.4) seem to be working fine.

```
import cdflib
fname = 'de2_ion2s_rpa_19830101_v01.cdf'
cdf = cdflib.CDF(fname)
cdf.varattsget('Epoch')
```
yields
```
File ~/miniconda3/lib/python3.10/site-packages/cdflib/cdfread.py:1344, in CDF._read_vdr2(self, byte_loc)
   1324 else:
   1325     compression_level = 0
   1326 return VDR(
   1327     data_type=data_type,
   1328     section_type=section_type,
   1329     next_vdr_location=next_vdr,
   1330     variable_number=var_num,
   1331     head_vxr=head_vxr,
   1332     last_vxr=last_vxr,
   1333     max_rec=max_rec,
   1334     name=name,
   1335     num_dims=num_dims,
   1336     dim_sizes=dim_sizes,
   1337     compression_bool=compression_bool,
   1338     compression_level=compression_level,
   1339     blocking_factor=blocking_factor,
   1340     dim_vary=dim_varys,
   1341     record_vary=record_variance_bool,
   1342     num_elements=num_elements,
   1343     sparse=sparse,
-> 1344     pad=pad,
   1345 )

UnboundLocalError: local variable 'pad' referenced before assignment
```

File obtained from https://cdaweb.gsfc.nasa.gov/pub/data/de/de2/plasma_rpa/ion2s_cdaweb/1983/

**Comment by jgieseler** [comment]
I ran into the same problem loading STEREO/HET data (which is old in that regard that it was established in 2006, but still delivers new data). Here is a working example using sunpy 4.1.7 and cdflib 1.0.2:

``` python
from sunpy.net import Fido
from sunpy.net import attrs as a
from sunpy.timeseries import TimeSeries
starttime="2022-1-26 10:30"
endtime="2022-1-26 15:40"
cda_dataset = a.cdaweb.Dataset('STA_L1_HET')
trange = a.Time(starttime, endtime)
result = Fido.search(trange, cda_dataset)
downloaded_file = Fido.fetch(result[0])
data = TimeSeries(downloaded_file, concatenate=True)
```

**Comment by dstansby** [comment]
Sorry for the issue, and thanks for the report - this is now fixed in 1.0.3, and a test added to make sure it doesn't break again.
